server/routes/README.md

```diff
@@ -0,0 +1,318 @@
+    "email": "email@email.com",
+    "firstName": "Foo",
+    "lastName": "Bar",
+    "password": "changeMePlz",
+    "createdAt": "2019-11-21T02:12:29.860Z",
+    "updatedAt": "2019-11-21T02:12:29.860Z"
+}
+```
+
+# Group Routes
+
+### Create Group
+To create a group with the current user as the admin
+
+    POST /api/group/create
+
+**Body Parameters**
+
+* groupName
+* description
+* visible - boolean `true | false`
+
+**Response**
+
+```
+{
+    "groupName": "MyGroup",
+    "admin": "foobar",
+    "description": "A group for friends to track their music progress",
+    "visible": true,
+    "members": [
+        "foobar",
+        "friendlyUser",
+        "papa.guitar72"
+    ],
+    "createdAt": "2019-11-21T02:12:29.860Z",
+    "updatedAt": "2019-11-21T02:12:29.860Z"
+}
+```
+
+### Get Group
+To get a group that is accessible to the current user
+
+    GET /api/group/{groupName}
+
+**Response**
+
+```
+{
+    "groupName": "MyGroup",
+    "admin": "foobar",
+    "description": "A group for friends to track their music progress",
+    "visible": true,
+    "members": [
+        "foobar",
+        "friendlyUser",
+        "papa.guitar72"
+    ],
+    "createdAt": "2019-11-21T02:12:29.860Z",
+    "updatedAt": "2019-11-21T02:12:29.860Z"
+}
+```
+
+### Add or Remove Group Member
+To add or remove a group member as the admin (as a member of the group?)
+
+    POST /api/group/{groupName}/add
+    POST /api/group/{groupName}/remove
+
+**Body Parameters**
+
+* username - of user to add or remove
+
+**Response**
+
+```
+{
+    "groupName": "MyGroup",
+    "admin": "foobar",
+    "description": "A group for friends to track their music progress",
+    "visible": true,
+    "members": [
+        "foobar",
+        "friendlyUser",
+        "papa.guitar72",
+        "mama.uke71"
+    ],
+    "createdAt": "2019-11-21T02:12:29.860Z",
+    "updatedAt": "2019-11-21T02:12:29.860Z"
+}
+```
+
+### Edit Group
+To edit something about a group as the admin
+
+    POST /api/group/{groupName}/edit
+
+**Body Parameters**
+
+* key - the name of the attribute to change, either `groupName | admin | description | visible`
+* value - the value to change the attribute to
+
+**Response**
+
+```
+{
+    "groupName": "MyGroup",
+    "admin": "foobar",
+    "description": "A group for friends to track their music progress",
+    "visible": true,
+    "members": [
+        "foobar",
+        "friendlyUser",
+        "papa.guitar72"
+    ],
+    "createdAt": "2019-11-21T02:12:29.860Z",
+    "updatedAt": "2019-11-21T02:12:29.860Z"
+}
+```
+
+### Delete Group
+To delete a group as the admin
+
+    POST /api/group/{groupName}/delete
+
+**Response**
+
+```
+{
+    "msg": "successfully deleted"
+}
+```
+
+# Recording Routes
+
+### Create Recording
+To create a recording as the current user
+
+    POST /api/recording/create
+
+**Body Parameters**
+
+* numPitches - an integer
+* instrument - a string
+* description - a string
+* start - a timestamp
+* end - a timestamp
```

Comment:
These are going to have to be named startTime and endTime, I believe 'start' is a special keyword in the request body